linera-service/tests/end_to_end_tests.rs

```diff
@@ -2334,7 +2336,28 @@ async fn test_project_publish() {
     let tmp_dir = client.project_new("init-test").await;
     let project_dir = tmp_dir.path().join("init-test");
 
-    client.project_publish(project_dir, vec![], None).await;
+    client.project_publish(project_dir, vec![], None, &()).await;
+    let chain = client.get_wallet().default_chain().unwrap();
+
+    let node_service = client.run_node_service(None).await;
+
+    assert_eq!(node_service.try_get_applications_uri(&chain).await.len(), 1)
+}
+
+#[test_log::test(tokio::test)]
+async fn test_example_publish() {
+    let _guard = INTEGRATION_TEST_GUARD.lock().await;
+
+    let network = Network::Grpc;
+    let mut runner = TestRunner::new(network, 1);
+    let client = runner.make_client(network);
+
+    runner.generate_initial_validator_config().await;
+    client.create_genesis_config().await;
+    runner.run_local_net().await;
+
+    let example_dir = TestRunner::example_path("counter");
+    client.project_publish(example_dir, vec![], None, &0).await;
```

Comment:
How do we know it used the workspace dir and didn't just build it in the `counter` folder?

Reply:
Because `cargo build` in the `counter` directory or the parent workspace always builds in `examples/target`.